sklearn/mixture/bernoulli_mixture.py

```diff
@@ -0,0 +1,268 @@
+"""Bernoulli Mixture Model."""
+
+# Author: Ben Dilday <ben.dilday@enigma.com>
+# License: BSD 3 clause
+
+import numpy as np
+
+from .base import BaseMixture
+from .gaussian_mixture import _check_means, _check_weights
+from ..utils.validation import check_is_fitted
+
+
+def log_no_underflow(x):
```

Comment:
Hmmmm I'd be somewhat surprised if we don't have something like this already. In any case the naming here implies it is part of our public API. Please prefix it with _.

Reply:
I wasn't able to find a similar function, so I added the underscore prefix.